### ISSUE TITLE
Vaporizer2 LV2: Parameter list doesn't appear

### ISSUE THREAD

**Issue by maxigaz** [BespokeSynth/BespokeSynth#1479]
## Describe the bug

The parameter list of Vaporizer2 LV2 v3.4.0 doesn't appear in Bespoke, so none of the parameters can be automated.

## Build info

bespoke NIGHTLY 1.2.1 (Oct 27 2023 02:49:21) - Linux - HEAD fd68fe3

## Steps To Reproduce

1. Load Vaporizer2 LV2.
2. Open the plugin's UI.
3. Move a knob of your choice, for example Custom modulator 1 on the Modulation Sources side panel. (Look for a button near the top right corner to toggle the side panel.)
4. In Bespoke, click on the drop down next to 'panic' with the label 'show parameters'.

## Expected vs. actual behavior

Nothing happens, even though the menu should appear with Custom modulator 1 in it.

**Comment by KottV** [comment]
Unfortunately it won't work due LV2 limitations: https://forum.juce.com/t/lv2-issues/52412/2
There is a PR with fix which still not accepted https://gitlab.com/lv2/lv2/-/merge_requests/29

So, using VST(3) is the only option here.

**Comment by maxigaz** [comment]
> So, using VST(3) is the only option here.

The VST3 version of Vaporizer2 doesn't even load in Bespoke.

![bespoke-vaporizer2-vst3](https://github.com/BespokeSynth/BespokeSynth/assets/10419911/8fa2f5ef-98fb-4741-b9af-4770cabcb5a1)

And here's what I'm seeing when I bring up Bespoke's developer console:

![bespoke-vaporizer2-vst3-console](https://github.com/BespokeSynth/BespokeSynth/assets/10419911/4ab84c2a-39c5-4e3c-8018-824d01727d85)

I would be willing to open a separate issue for that, but I don't know if it belongs to Bespoke, JUCE, or the plugin itself. What do you think, @KottV?

**Comment by KottV** [comment]
@maxigaz where are Bespoke and Vaporizer2 installed from, and what is your OS?

**Comment by KottV** [comment]
@maxigaz got this on my system too, it was caused by changing Vaporizer2 placement (from ~/.vst3 to /usr/lib64/vst3). 
If it's your case, to resolve this:
1. Remove all VST3 Vaporizer2 from Bespoke's plugin manager list
2. Clean up `~/Documents/BespokeSynth/vst/recent_plugins.json` or remove it.
3. Re-scan VST3 plugins in Bespoke's plugin manager.

**Comment by maxigaz** [comment]
> @maxigaz where are Bespoke and Vaporizer2 installed from, and what is your OS?

Bespoke is the nightly build downloaded from [here](https://github.com/BespokeSynth/BespokeSynth/releases/tag/Nightly), Vaporizer2 is from the Ubuntu package hosted [here](https://software.opensuse.org/download.html?project=multimedia%3Aproaudio&package=vaporizer2-vst3). (For the record, I also tried compiling Vaporizer2 from the main branch myself about a week ago, and I had the exact same issue with the VST3 version.) The OS is Arch Linux.

> @maxigaz got this on my system too, it was caused by changing Vaporizer2 placement (from ~/.vst3 to /usr/lib64/vst3). If it's your case, to resolve this:
> 
>     1. Remove all VST3 Vaporizer2 from Bespoke's plugin manager list
> 
>     2. Clean up `~/Documents/BespokeSynth/vst/recent_plugins.json` or remove it.
> 
>     3. Re-scan VST3 plugins in Bespoke's plugin manager.

I've had it in `~/.vst3` all along. I tried these steps just in case, but it still doesn't load in Bespoke.

**Comment by KottV** [comment]
>I also tried compiling Vaporizer2 from the main branch myself about a week ago 

do you mean 3.4.0 release? it was merged 4 days ago

anyway, does Vaporizer2 VST3 works in any other host, like Carla?

**Comment by maxigaz** [comment]
> do you mean 3.4.0 release? it was merged 4 days ago

No, I mean the main branch on GitHub. I had tried it just ~a few days~ before I learned that 3.4.0 was out. (Correction: It was actually on the same day, a few hours before the release.)

> anyway, does Vaporizer2 VST3 works in any other host, like Carla?

Yes, I can confirm that it works in Carla.

**Comment by KottV** [comment]
Ok, this is good.
Does it looks like that in your BespokeSynth PM?

![изображение](https://github.com/BespokeSynth/BespokeSynth/assets/4686351/d24bd3bb-7b23-4935-a48a-422d9df255ed)

I still believe there is another old installation or plugins list entrance.

**Comment by maxigaz** [comment]
> Ok, this is good. Does it looks like that in your BespokeSynth PM?
> 
> ![изображение](https://private-user-images.githubusercontent.com/4686351/288476685-d24bd3bb-7b23-4935-a48a-422d9df255ed.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE4ODM1NjAsIm5iZiI6MTcwMTg4MzI2MCwicGF0aCI6Ii80Njg2MzUxLzI4ODQ3NjY4NS1kMjRiZDNiYi03YjIzLTQ5MzUtYTQ4YS00MjJkOWRmMjU1ZWQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMDZUMTcyMTAwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Yzg0ODhkZmEyNjY0ZjY0MGJjMjQ5NzlhODMzYjk3ZjNhMDM0NDk0ODkzMWEyMzgxNTY5OGRlM2Y5OGU2ODg2ZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.BnHUFMzVSNQ-cJOpe74cSImKdszVtgpdE-WmBcrIzPM)

Yes, except I don't have a version number for Vaporizer2 (VST3) in the 'Description' column.

**Comment by KottV** [comment]
But it should have this number if plugin is found and present.

**Comment by crshrprt** [comment]
I have the same issue with the VST3 on Debian Sid. Done all the above steps + removed the configuration in `~/.config/Vaporizer2` just to be sure. Vaporizer2 is found when scanning but is listed without the version number.

**Comment by KottV** [comment]
@maxigaz @crshrprt what is exact Ubuntu (or Debian) version of your Vaporizer2 package?

**Comment by KottV** [comment]
IDK where the bug exactly
as a workaround delete  `VASTvaporizer2.vst3/Contents/Resources/moduleinfo.json` file
also you can find vaporizer2-vst  in the same OBS repo, which shouldn't be affected
also, if you on Debian 11 or Ubuntu 20, there is a little issue with build in OBS, so get Vaporizer2 here https://github.com/KottV/Vaporizer2/releases/tag/3.4.3 (in Assets there are Ubuntu 20 builds)

**Comment by crshrprt** [comment]
@KottV I used the Debian 12 `vaporizer2-vst3` and `vaporizer2-data` package from [here](https://software.opensuse.org/download.html?project=multimedia%3Aproaudio&package=vaporizer2-vst3). I have the same problem with VST3 if I build it myself. 
Indeed removing `VASTvaporizer2.vst3/Contents/Resources/moduleinfo.json` file make it work in BespokeSynth (and it shows version number in the Plugin Manager window), thank you.

**Comment by maxigaz** [comment]
> what is exact Ubuntu (or Debian) version of your Vaporizer2 package?

@KottV I used the package for Ubuntu 23.10.

> as a workaround delete `VASTvaporizer2.vst3/Contents/Resources/moduleinfo.json`

I can confirm that it works. It now loads in Bespoke. Thank you!

> get Vaporizer2 here https://github.com/KottV/Vaporizer2/releases/tag/3.4.3 (in Assets there are Ubuntu 20 builds)

I've also just tried that one, and `moduleinfo.json` also needs to be deleted or renamed for it to load.

**Comment by KottV** [comment]
problem is in JUCE that BespokeSynth is build on https://github.com/juce-framework/JUCE/commit/e3b14702a066e6ee3100f057de6f47673d629e0f
this fix is added to Bepoke's OBS builds (some of them are red, but I'm working on it)  https://build.opensuse.org/package/show/home:bespokesynth/bespokesynth-nightly https://build.opensuse.org/package/show/home:bespokesynth/bespokesynth

**Comment by NoxyNixie** [comment]
I believe this is then resolved right? If not you can reopen or reply here and I'll reopen.

**Comment by maxigaz** [comment]
I can still reproduce the original issue in Bespoke Nightly (Linux - main dcb6a4c8) with Vaporizer 3.5.0 LV2 (downloaded from KottV's repo [here](https://github.com/KottV/Vaporizer2/releases/tag/3.5.0)).

However, at least the VST3 version (KottV's build) seems to work fine.

**Comment by NoxyNixie** [comment]
Can you reproduce it in another JUCE host? If it works in another JUCE host we may have to update JUCE else JUCE still needs to fix the issue.

**Comment by maxigaz** [comment]
> Can you reproduce it in another JUCE host?

I'm not familiar with any other piece of software that uses JUCE for hosting LV2 plugins. If you have any suggestions (preferably something that is free and open source and is Linux-native), I might be able to give it a try.
